Submitted & accepted upstream in:
https://gitlab.freedesktop.org/dbus/dbus/-/merge_requests/478


From 7b3d0fd3cdc1cfea905ef814b872e9976b609d95 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Mon, 7 Oct 2024 17:49:17 -0700
Subject: [PATCH] configure: Fix check for Solaris audit API to work with gcc
 14

Need to declare a variable type to avoid gcc 14 error:

configure:22502: checking for ADT API
conftest.c:107:1: warning: data definition has no type or storage class
  107 | adt_user_context = ADT_USER;
      | ^~~~~~~~~~~~~~~~
conftest.c:107:1: error: type defaults to 'int' in declaration of
 'adt_user_context' [-Wimplicit-int]

Fixes: ab1eb1fd ("Bug 15740: Solaris/ADT auditing support (simon zheng)")
Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ecbdf2a3c..d258f8cb6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1042,7 +1042,7 @@ AC_SUBST([SELINUX_LIBS])
 AC_MSG_CHECKING(for ADT API)
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include <bsm/adt.h>
-adt_user_context = ADT_USER;
+int adt_user_context = ADT_USER;
 ]], [[]])], [ check_adt_audit=yes ], [ check_adt_audit=no ])
 
 if test ${check_adt_audit} = yes
-- 
GitLab

